internal/bminventory/inventory_test.go

```diff
@@ -71,6 +72,12 @@ func getTestAuthHandler() auth.AuthHandler {
 	return *auth.NewAuthHandler(fakeConfigDisabled, nil, getTestLog().WithField("pkg", "auth"), nil)
 }
 
+func getPullSecretValidator(ctrl *gomock.Controller) validations.PullSecretValidator {
+	m := validations.NewMockPullSecretValidator(ctrl)
+	m.EXPECT().ValidatePullSecret(gomock.Any(), gomock.Any(), gomock.Any()).Return(nil).AnyTimes()
```

Comment:
Why `AnyTimes`? isn't `AnyTimes` can be zero?

Reply:
In this context it needs to be just a stub. There is already a suite that focuses on the validation functionality, I don't see value in painstakingly mocking this for each of the other tests in this module. So it always passes - as before, but now as an injected object.

Reply:
My concern is that it can hide bugs, if there are tests that validate it specifically then maybe adding a comment that it's a stub will help, other wise someone can just delete this call and all the tests will pass.

Reply:
I understand that, but before my changes it was a static call which always received valid input. We didn't care if it was actually called or not - it always passed. Now it's an object, so now even for tests that do not focus on pull secret validation it just needs to be there. I used an instance that behaved exactly as before (i.e. passed when called, because the input was always valid), but you suggested that I use a mock. I think a mock (actually a stub) is indeed better because it is explicit, but its role is still just to allow for testing the other functionality without dealing with pull secret validation. If I start tailoring the expectations of this mock to every existing test case, I'll have to fix 150+. Each of them may call the stub, or may not, or may call it multiple times, but none of them care about this functionality. I prefer to add a dedicated test suite instead.